pkg/iac/adapters/cloudformation/aws/s3/s3_test.go

```diff
@@ -0,0 +1,108 @@
+package s3
+
+import (
+	"context"
+	"testing"
+
+	"github.com/aquasecurity/trivy/internal/testutil"
+	"github.com/aquasecurity/trivy/pkg/iac/providers/aws/s3"
+	"github.com/aquasecurity/trivy/pkg/iac/scanners/cloudformation/parser"
+	"github.com/aquasecurity/trivy/pkg/iac/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAdapt(t *testing.T) {
```

Comment:
Thanks for adding the test as we severely lack coverage for cloudformation and can improve it. Could you also add an unhappy path where maybe the algorithm isn't a valid one and further convert these two cases into a table driven test?

Reply:
Done https://github.com/aquasecurity/trivy/pull/6270/commits/69f7e4cd4ba60d71c008a34ed7086164de2522a8